File: 프로그래머스/2/131127. 할인 행사/할인 행사.js
function solution(want, number, discount) {
    const map = new Map();
    let result = 0;
    for(let i = 0; i < want.length; i++){
        map.set(want[i], number[i]);
    }
    for(let i = 0; i <= discount.length - 10; i++){
        let newmap = new Map(map);
        let product = "";
        let count = 0
        for(let j = i; j < i+10; j++){
            product = discount[j]
            count = newmap.get(product)
            if(count > 0){
                if(count === 1) {
                    newmap.delete(product);
                }
                else {
                    newmap.set(product, count-1)
                }
            }
        }
        if(newmap.size === 0) result ++;
    }
    return result
}

File: 프로그래머스/2/131127. 할인 행사/할인 행사.py
def solution(want, number, discount):
    dic = {}
    result = 0
    for i in range(len(want)):
        dic[want[i]] = number[i]
        
    for i in range(len(discount) - 10 + 1):
        new_dic = {}
        for j in range(10):
            if discount[i+j] in new_dic:
                new_dic[discount[i+j]] += 1
            else:
                new_dic[discount[i+j]] = 1
        if dic == new_dic:
            result += 1
            
    return result